scripts/globals/events/login_campaign.lua

```diff
@@ -0,0 +1,230 @@
+------------------------------------
+-- Login Campaign
+------------------------------------
+require("scripts/globals/npc_util")
+------------------------------------
+
+tpz = tpz or {}
+tpz.events = tpz.events or {}
+tpz.events.loginCampaign = tpz.events.loginCampaign or {}
+
+-- Change vars below to modify settings for current login campaign
+loginCampaignYear = 2020
+loginCampaignMonth = 12
+loginCampaignDay = 10
+loginCampaignDuration = 23 -- Duration is set in Earth days (Average is 23 days)
+
+-- Checks if a Login Campaign is active.
+tpz.events.loginCampaign.isCampaignActive = function()
+    if ENABLE_LOGIN_CAMPAIGN == 1 then
+        local currentLocalTime = os.time(os.date('*t'))
+        local currentUTCTime = os.time(os.date('!*t'))
+        local jstOffset = currentUTCTime - currentLocalTime + 32400
+        local campaignStartDate = os.time({
+            year = loginCampaignYear,
+            month = loginCampaignMonth,
+            day = loginCampaignDay,
+            hour = 0,
+            min = 0,
+            sec = 0
+        }) + jstOffset  -- JST time
```

Comment:
I believe this can be shortened to just
```lua
        local campaignStartDate = os.time({
            year = loginCampaignYear,
            month = loginCampaignMonth,
            day = loginCampaignDay,
            hour = 0,
            min = 0,
            sec = 0
        }) + 9 * 60 * 60  -- JST time
 ```
 `os.time(os.date('*t'))` will give you the same result as `os.time()`, and `os.date('!*t')` is getting a table for the current date in UTC, then `os.time()` is getting a timestamp of that adjusted date as if you were asking what time UTC it will be when it is that date in _your_ timezone, which is actually UTC+(diff between your time and UTC). So when you subtract `currentLocalTime` you're just setting `currentUTCTime` back to `os.time()`.

Reply:
Awesome and it makes it more clear, love the use of underscoring here since cammel naming woulda made it messy.
Made the changes, thank you coco <3